src/datasources/balances-api/safe-balances-api.service.ts

```diff
@@ -198,6 +198,11 @@ export class SafeBalancesApi implements IBalancesApi {
         const fiatBalance = this._getFiatBalance(price, balance);
         return {
           ...balance,
+          token: {
+            ...balance.token,
+            decimals:
+              balance.token?.decimals ?? SafeBalancesApi.DEFAULT_DECIMALS,
+          },
```

Comment:
As there are [multiple](https://github.com/safe-global/safe-client-gateway/blob/decimals/src/datasources/balances-api/zerion-balances-api.service.ts#L46) `IBalanceApi`s, we should assign the default value [during validation](https://github.com/safe-global/safe-client-gateway/blob/decimals/src/domain/tokens/entities/schemas/token.schema.ts#L8). It then doesn't need to be done in both places and will solve [the `MultisigTransactionExecutionDetailsMapper` CI error](https://github.com/safe-global/safe-client-gateway/actions/runs/12952281091/job/36129259361?pr=2291#step:6:14).

Reply:
Nice, thanks for the pointer!